src/backend/core/api/viewsets.py

```diff
@@ -686,15 +737,35 @@ class TemplateViewSet(
 ):
     """Template ViewSet"""
 
+    filter_backends = [filters.OrderingFilter]
     permission_classes = [
         permissions.IsAuthenticatedOrSafe,
         permissions.AccessPermission,
     ]
+    ordering = ["-created_at"]
+    ordering_fields = ["created_at", "updated_at", "title"]
     serializer_class = serializers.TemplateSerializer
-    access_model_class = models.TemplateAccess
-    resource_field_name = "template"
     queryset = models.Template.objects.all()
 
+    def get_queryset(self):
+        """Custom queryset to get user related templates."""
+        queryset = super().get_queryset()
+        user = self.request.user
+
+        if not user.is_authenticated:
+            return queryset
```

Comment:
Should you filter on public templates only? Because like this you may be able to get any template anonymously

Reply:
I'm not trying to handle permissions here because it's done in the permission class : 
This was done following one of your feedbacks a few weeks ago where you requested to split more clearly what is permission and what is annotations (queryset preparation).

More specifically, this is the code that will block anonymous users from accessing any template: https://github.com/numerique-gouv/impress/blob/main/src/backend/core/models.py#L679

I'm confident that anonymous users can't retrieve any template thanks to the test suite e.g https://github.com/numerique-gouv/impress/blob/main/src/backend/core/tests/templates/test_api_templates_retrieve.py#L38

Reply:
True ^^